galley/pkg/config/resource/name.go

```diff
@@ -52,6 +57,18 @@ func NewFullName(name string) (Name, error) {
 	return Name{string: name}, nil
 }
 
+// NewFullOrLocalName returns a given name as a resource Name.  If name was a local name, the supplied namespace is used
+func NewFullOrLocalName(namespace, name string) Name {
```

Comment:
I like `NewShortOrFullName`. "Short" name is consistent with how we describe these in the docs.

Can we also use `defaultNamespace` instead of `namespace` to make it more clear how that argument is used?

Reply:
I was going to comment the same. For reader, it doesn't directly notify that the ``namespace`` attribute might be ignored and something else than potentially intended might be returned.